packages/dd-trace/src/opentracing/span.js

```diff
@@ -158,16 +158,27 @@ class DatadogSpan {
         trace: parent._trace,
         tracestate: parent._tracestate
       })
+
+      if (!spanContext._trace.startTime) {
+        spanContext._trace.startTime = dateNow()
+      }
     } else {
       const spanId = id()
+      const startTime = dateNow()
       spanContext = new SpanContext({
         traceId: spanId,
         spanId
       })
+      spanContext._trace.startTime = startTime
+
+      if (fields.traceId128BitGenerationEnabled) {
+        spanContext._trace.tags['_dd.p.tid'] = Math.floor(startTime / 1000).toString(16)
+          .padStart(8, '0')
+          .padEnd(16, '0')
+      }
     }
 
     spanContext._trace.started.push(this)
-    spanContext._trace.startTime = spanContext._trace.startTime || dateNow()
```

Comment:
is this change unrelated?

Reply:
No, this was moved to each conditional branch since now the logic is different between when there is a parent and not.